--- texlive-20080816-source/libs/icu-xetex/tools/genuca/genuca.cpp.orig	2008-03-04 14:26:36.000000000 +0100
+++ texlive-20080816-source/libs/icu-xetex/tools/genuca/genuca.cpp	2017-12-06 08:54:13.824347675 +0100
@@ -882,7 +882,7 @@
 struct {
       UChar32 start;
       UChar32 end;
-      int32_t value;
+      uint32_t value;
     } ranges[] =
     {
 #if 0
--- texlive-20080816-source/libs/graphite-engine/src/segment/Segment.cpp~	2008-03-18 12:29:11.000000000 +0100
+++ texlive-20080816-source/libs/graphite-engine/src/segment/Segment.cpp	2017-12-07 09:32:35.371698201 +0100
@@ -570,6 +570,7 @@
 	Swap the guts of the two segments.
 	OBSOLETE
 ----------------------------------------------------------------------------------------------*/
+/*
 void Segment::SwapWith(Segment * pgrseg)
 {
 	int crefThis = m_cref;
@@ -580,7 +581,7 @@
 	m_cref = crefThis;
 	pgrseg->m_cref = crefOther;
 }
-
+*/
 /*----------------------------------------------------------------------------------------------
 	Delete the pointer to the alternate-end-of-line segment.
 ----------------------------------------------------------------------------------------------*/
