--- texlive-20091011-source/./texk/xdvipdfmx/src/xbb.c~	2008-05-25 17:00:35.000000000 +0200
+++ texlive-20091011-source/./texk/xdvipdfmx/src/xbb.c	2010-01-28 22:01:10.653876887 +0100
@@ -246,7 +246,7 @@
   unsigned char sigbytes[4];
   rewind (png_file);
   if (fread(sigbytes, 1, sizeof(sigbytes), png_file) != sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes)))) return 0;
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes)))) return 0;
   else return 1;
 }             
 
--- texlive-20091011-source/texk/xdvipdfmx/src/pngimage.c~	2007-11-22 17:48:06.000000000 +0100
+++ texlive-20091011-source/texk/xdvipdfmx/src/pngimage.c	2010-01-28 22:02:06.753257437 +0100
@@ -146,7 +146,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes))))
     return 0;
   else
     return 1;
--- texlive-20091011-source/./texk/dvipdfmx/src/pngimage.c~	2009-05-26 01:12:24.000000000 +0200
+++ texlive-20091011-source/./texk/dvipdfmx/src/pngimage.c	2010-01-28 22:02:43.449910278 +0100
@@ -148,7 +148,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes))))
     return 0;
   else
     return 1;
--- texlive-20091011-source/./texk/web2c/xetexdir/pngimage.c~	2006-07-26 18:19:45.000000000 +0200
+++ texlive-20091011-source/./texk/web2c/xetexdir/pngimage.c	2010-01-28 22:03:20.986780607 +0100
@@ -101,7 +101,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes))))
     return 0;
   else
     return 1;
