--- texlive-20080816-source-orig/texk/lcdf-typetools/otftotfm/Makefile.in	2008-05-14 11:50:18.000000000 +0200
+++ texlive-20080816-source/texk/lcdf-typetools/otftotfm/Makefile.in	2008-12-14 11:39:10.000000000 +0100
@@ -99,8 +99,8 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 KPSEINTERFACE_O = @KPSEINTERFACE_O@
 LDFLAGS = @LDFLAGS@
-LIBKPATHSEA = @LIBKPATHSEA@
-LIBKPATHSEA_DEP = @LIBKPATHSEA_DEP@
+LIBKPATHSEA = $(top_builddir)/../../texk/kpathsea/.libs/libkpathsea.so
+LIBKPATHSEA_DEP = $(top_builddir)/../../texk/kpathsea/.libs/libkpathsea.so
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LTLIBOBJS = @LTLIBOBJS@
